[IMP] barcode: overhaul inventory adj (#12461)
odoo/documentation: content/applications/inventory_and_mrp/barcode/operations/adjustments.rst

.. tip::
   Odoo's **Barcode** application provides demo data with barcodes to explore the features of the
   app. These can be used for testing purposes, and can be printed from the home screen of the app.

   To access this demo data, navigate to the :menuselection:`Barcode app` and click :guilabel:`stock
   barcodes sheet` and :guilabel:`commands for Inventory` in the information pop-up window above the
   scanner.

   .. image:: adjustments/adjustments-barcode-stock-sheets.png
      :alt: Demo data prompt pop-up on Barcode app main screen.

Assigning inventory counts
==========================

Before performing an inventory count, managers can :ref:`assign <inventory/plan-counts>` counting
tasks to employees. This can be done through the **Inventory** app by selecting specific locations
and products for counting. Once assigned, users will see pending counts when they open the
**Barcode** app.

To view a requested inventory count, navigate to the :menuselection:`Barcode app` dashboard. If a
count has been requested, the number of products to be counted is listed on the :guilabel:`Inventory
Count` button.

.. image:: adjustments/assigned-count.png

>    :alt: The Barcode dashboard with an assigned count.

Author: This is what the screen looks like when there are assigned counts (at least in the barcode app). I just ran another test in inventory, requested a count from Mitchell and went to the barcode app and the image is exactly the same. Are you seeing something different?

>    :alt: The Barcode dashboard with an assigned count.

Reviewer: I see a little number appear in the bubble

![image](https://github.com/user-attachments/assets/6b8f7677-a464-4611-8eef-58c8f06e16e7)

:guilabel:`Confirm` on the pop-up.

.. danger::
   Enabling the **Barcode** feature requires installing the **Barcode** application. Installing a
   new application on a One-App-Free database triggers a 15-day trial. At the end of the trial, if a
   paid subscription has not been added to the database, it will no longer be accessible.

After saving, a new drop-down menu appears under the :guilabel:`Barcode Scanner` option, labeled
:guilabel:`Barcode Nomenclature`, where either :guilabel:`Default Nomenclature` or
:guilabel:`Default GS1 Nomenclature` can be selected. Each nomenclature option determines how
scanners interpret barcodes in Odoo.

To count products using barcodes, ensure that barcodes for products and storage locations are set up
within Odoo first. Refer to this section for detailed instructions: :ref:`Set Product Barcodes
<inventory/barcode/set-barcodes>`.

.. image:: adjustments/adjustments-barcode-setting.png
   :alt: Enabled Barcode feature in Inventory app settings.

.. _inventory/barcode/perform-count:

Performing an inventory count
=============================

To perform an inventory adjustment, first navigate to the :menuselection:`Barcode app`. If assigned

> counts exist,tap :guilabel:`Inventory Count` to view pending tasks.

Reviewer: I think we should add a space

![image](https://github.com/user-attachments/assets/e0d575ed-15df-4108-a29c-06dc20e9ab52)

scanner, or the Odoo mobile app.

.. note::
   For a list of Odoo-compatible barcode mobile scanners, and other hardware for the **Inventory**
   and **Barcode** apps, refer to the `Odoo Inventory • Hardware page
   <https://www.odoo.com/app/inventory-hardware>`_.

.. seealso::
   :doc:`../../inventory/warehouses_storage/inventory_management/count_products`

.. tip::
   Odoo's **Barcode** application provides demo data with barcodes to explore the features of the
   app. These can be used for testing purposes, and can be printed from the home screen of the app.

   To access this demo data, navigate to the :menuselection:`Barcode app` and click :guilabel:`stock
   barcodes sheet` and :guilabel:`commands for Inventory` in the information pop-up window above the
   scanner.

   .. image:: adjustments/adjustments-barcode-stock-sheets.png
      :alt: Demo data prompt pop-up on Barcode app main screen.

Assigning inventory counts
==========================

Before performing an inventory count, managers can :ref:`assign <inventory/plan-counts>` counting
tasks to employees. This can be done through :menuselection:`Inventory app --> Operations -->

> Physical Inventory` by selecting specific locations and products for counting, and assigning a User

Reviewer: ```suggestion
Physical Inventory` by selecting specific locations and products for counting, and assigning a :guilabel:`User`
```

to them. Once assigned, users will see pending counts when they open the **Barcode** app.

To view a requested inventory count, navigate to the :menuselection:`Barcode app` dashboard. If a
count has been requested, the number of products to be counted is listed on the :guilabel:`Inventory

> Count` button.

Reviewer: ```suggestion
count` button.
```

Adjust inventory with barcodes
==============================

An *inventory adjustment*, or inventory audit, is the process of verifying the physical stock of
products against the quantities recorded in the database. Regular audits ensure accurate inventory
records, prevent stock discrepancies, and maintain efficient operations. In a warehouse setting,
managers typically assign inventory counts to employees, who then walk to designated locations, scan
product barcodes, and adjust quantities as needed.

Inventory adjustments can be completed through the **Barcode** application using a compatible
scanner, or the Odoo mobile app.

.. note::
   For a list of Odoo-compatible barcode mobile scanners, and other hardware for the **Inventory**
   and **Barcode** apps, refer to the `Odoo Inventory • Hardware page
   <https://www.odoo.com/app/inventory-hardware>`_.

.. seealso::
   :doc:`../../inventory/warehouses_storage/inventory_management/count_products`

.. tip::
   Odoo's **Barcode** application provides demo data with barcodes to explore the features of the
   app. These can be used for testing purposes, and can be printed from the home screen of the app.

>    To access this demo data, navigate to the :menuselection:`Barcode app` and click :guilabel:`stock
>    barcodes sheet` and :guilabel:`commands for Inventory` in the information pop-up window above the
>    scanner.

Reviewer 2: The UI and image are showing links "demo data sheet" and "barcode" not "stock barcodes sheet" or "commands for Inventory"
```suggestion
   To access this demo data, navigate to the :menuselection:`Barcode app` and click :guilabel:`demo
   data sheet` or :guilabel:`barcodes` in the banner above the scanner.
```

----------------------

The :guilabel:`Count Entire Locations` feature allows users to count all of the products within a
specific location systematically. When scanning a location for an inventory count, the user is
assigned to count all of the products within that location. This allows for easier cycle counts by
assigning an entire location to a user by assigning a single product count. During cycle counts,
users can ensure accurate inventory numbers, see if products that should be in a location are
missing, or discover products incorrectly stored within a location.

To enable this feature, navigate to :menuselection:`Inventory app --> Configuration --> Settings`,
and scroll to the :guilabel:`Barcode` section. Tick the :guilabel:`Count Entire Locations` checkbox,
then click :guilabel:`Save`.

.. important::
   This setting is only visible if the :guilabel:`Storage Locations` checkbox is ticked.

To perform an inventory cont of an entire location, navigate to :menuselection:`Barcode app -->
Inventory Count`,  then scan the desired location barcode. Doing so displays all the products
assigned to the location. :ref:`Proceed with the count <inventory/barcode/conduct-adjustment>` as

> normal.

Reviewer: I’m a bit unsure about the placement of this paragraph. It’s currently located in the "Preparing" section, which is followed by a "Conduct Adjustment" section. However, the instructions in this paragraph seem to focus on how to conduct  the inventory adjustment rather than on the preparation steps. 

To improve the overall clarity and flow of the documentation, let’s keep the "Preparing for an Inventory Adjustment" section focused solely on setup tasks, such as configuring settings, assigning counts, or organizing resources. The instructions for actually conducting the adjustment would then fit better in the "Conduct Adjustment" section. 

Would it make sense to move this paragraph to the "Conduct Adjustment" section?

Show quantity to count
----------------------

When conducting an inventory count, the expected quantity of products is displayed by default, to
provide the user with a baseline to use when performing the count. However, as this can result in
users relying on this count instead of performing a new count, this quantity can be hidden.

Navigate to :menuselection:`Inventory app --> Configuration --> Settings`. In the
:guilabel:`Barcode` section, clear the :guilabel:`Show Quantity to Count` checkbox, then click
:guilabel:`Save`.

.. image:: adjustments/show-quantity-to-count-disabled.png
   :alt: An inventory count without the show quantity to count feature enabled.

.. _inventory/barcode/conduct-adjustment:

> Conducting an inventory adjustment

Reviewer: Thank you for putting this section together! I think we can make it even more helpful by adding a bit more context to guide the user through the process. (It's completely okay that this wasn't included initially—this is more of an inventory and supply chain concept, and I’m happy to help provide some clarity!) 

To give users a better sense of how to use the Barcode app in real-life scenarios, let’s acknowledge the practical elements of counting products in a warehouse setting. For example, we can describe how the user interacts with the app while physically navigating the warehouse. You might find it helpful to refer to the [Barcode tour video ](https://www.youtube.com/watch?v=2pwXDP_LQGA&t=3s) for a deeper understanding of the use case. 

Here’s the current intro:

> Navigate to the Barcode app ‣ Inventory count. To begin the adjustment, first scan the source location, which is the current location in the warehouse of the product whose count should be adjusted. Then, scan the product barcodes.

Let’s expand on this to include more context about the user’s experience and the steps they’ll take. For example:  
- If inventory counts have been assigned to the user, they’ll see the number of counts displayed on the Inventory counts  button when they open the Barcode app. (While reviewing this, I noticed that the document doesn’t currently explain how to assign inventory counts to a user. Could we double-check if this is covered elsewhere, such as in the inventory adjustment documentation? If not, it might be worth adding.)  
- Walk the user through the process step-by-step: Where is the storage location? Should they walk there first? Once they arrive, they’ll scan the location—what happens in the UI when they do that?  
- After scanning the location, they’ll scan the product barcode. Explain how they can manually adjust quantities and what feedback the app provides during this process.
   
By including these details, we’ll give users a much clearer picture of how to use the app effectively in a real-world scenario. For example, here’s how the revised intro might look:  

> Navigate to the Barcode app. If inventory counts have been assigned to you, the number of counts will be displayed on the Inventory counts  button. Each count corresponds to a specific storage location in the warehouse. Begin by walking to the assigned storage location. Once there, scan the location barcode to confirm your position—the app will display the products stored in that location. Next, scan the product barcodes and verify or adjust the quantities as needed. 

Does this approach make sense? Let me know if you’d like to discuss further or need help refining any part of this!

> Apply inventory adjustments with barcodes

Reviewer 3: ```suggestion
Adjust Inventory with barcodes
```
Not sure if  that's correct but it's shorter

Inventory adjustments can be completed through the **Barcode** application using a compatible
scanner, or the Odoo mobile app.

.. note::
   For a list of Odoo-compatible barcode mobile scanners, and other hardware for the **Inventory**
   and **Barcode** apps, refer to the `Odoo Inventory • Hardware page
   <https://www.odoo.com/app/inventory-hardware>`_.

.. seealso::
   :doc:`../../inventory/warehouses_storage/inventory_management/count_products`

.. tip::

   Odoo's **Barcode** application provides demo data with barcodes to explore the features of the
   app. These can be used for testing purposes, and can be printed from the home screen of the app.

   To access this demo data, navigate to the :menuselection:`Barcode app` and click :guilabel:`stock
   barcodes sheet` and :guilabel:`commands for Inventory` in the information pop-up window above the
   scanner.

   .. image:: adjustments/adjustments-barcode-stock-sheets.png
      :alt: Demo data prompt pop-up on Barcode app main screen.

Preparing for an inventory adjustment
=====================================

Before an inventory adjustment can be performed with the **Barcode** app, the app has to be

> installed, and configured. Navigate to the :menuselection:`Inventory app --> Configuration -->
> Settings --> Barcode`. Tick the checkbox next to :guilabel:`Barcode Scanner`. Click :guilabel:`Save`
> to save the changes. If necessary, click :guilabel:`Confirm` on the pop-up.

Reviewer 3: ```suggestion
installed, and configured. Navigate to the :menuselection:`Inventory app --> Configuration -->
Settings`, and scroll to the :guilabel:'Barcode' section. Tick the checkbox next to :guilabel:`Barcode Scanner`, and click :guilabel:`Save`
to save any changes. If necessary, click :guilabel:`Confirm` on the pop-up.
```

.. danger::
   Enabling the **Barcode** feature requires installing the **Barcode** application. Installing a
   new application on a One-App-Free database triggers a 15-day trial. At the end of the trial, if a
   paid subscription has not been added to the database, it will no longer be accessible.

After saving, a new drop-down menu appears under the :guilabel:`Barcode Scanner` option, labeled
:guilabel:`Barcode Nomenclature`, where either :guilabel:`Default Nomenclature` or
:guilabel:`Default GS1 Nomenclature` can be selected. Each nomenclature option determines how
scanners interpret barcodes in Odoo.

Below this is a :icon:`oi-arrow-right` :guilabel:`Configure Product Barcodes` internal link, along

> with a set of :guilabel:`Print` buttons for printing barcode commands and a barcode demo sheet.

Reviewer 3: ```suggestion
with a set of :guilabel:`Print` buttons for printing barcode commands, storage locations, and a barcode demo sheet.
```

.. image:: adjustments/adjustments-barcode-setting.png
   :alt: Enabled Barcode feature in Inventory app settings.

.. seealso::
   For more information on setting up and configuring the **Barcode** app, refer to the :doc:`Set up
   your barcode scanner <../setup/hardware>` and :doc:`Activate the Barcodes in Odoo
   <../setup/software>` docs.

Count entire locations
----------------------

The :guilabel:`Count Entire Locations` feature allows users to count all of the products within a
specific location systematically. When scanning a location for an inventory count, the user is

> assigned to count all of the products within that location. This allows users to see if products
> that should be in a location are missing.

Reviewer 3: ```suggestion
assigned to count all of the products within that location. This allows for easier cycle counts by assigning an entire location to a user by assigning a single product count. During cycle counts, users can ensure accurate inventory numbers, see if products
that should be in a location are missing, or discover products incorrectly stored within a location.
```

I just wanted to imply that this is not JUST for seeing if things are missing. I was thinking this through, and it could be a really easy way to do a cycle count (what we called check-in inventory) periodically, because you basically are assigning a wholelocation with ONE product- which is amazing. You don't need to do all the extra clicks to assign other stuff. Please reword this - this was just my first try.

Also, I recommend adding an example field where you can flesh it out. So you can say 'We have lots of shelves with lots of locations, and need each location to be checked by an employee. Scanning the first item in the first shelf for a location assigns.... etc'. Because I THINK this is to save time by tailoring cycle counts? You just assign one item and it assigns it all for the location>? So you can set up your team of inventory checkers with one scanning of one product- very cool. So yeah, I tink an nice example block with a use case could be awesome here to really illustrate why anyone would use this feature,

> To enable this feature, navigate to :menuselection:`Inventory app --> Configuration --> Settings`.
> Tick the :guilabel:`Count Entire Locations` checkbox, then click :guilabel:`Save`.

Reviewer 3: ```suggestion
To enable this feature, navigate to :menuselection:`Inventory app --> Configuration --> Settings`, and scroll to the :guilael:' Barcode' section.
Tick the :guilabel:`Count Entire Locations` checkbox, then click :guilabel:`Save`.
```

.. important::
   This setting is only visible if the :guilabel:`Storage Locations` checkbox is ticked.

> To perform an inventory cont of an entire location, navigate to :menuselection:`Barcode app -->
> Inventory Count`. Scan the location barcode. Doing so results in all of the products assigned to the

Reviewer 3: ```suggestion
To perform an inventory cont of an entire location, navigate to :menuselection:`Barcode app -->
Inventory Count`,  then scan the desired location barcode. Doing so displays all the products assigned to the
```

> location being displayed. :ref:`Proceed with the count <inventory/barcode/conduct-adjustment>` as

Reviewer 3: ```suggestion
location. :ref:`Proceed with the count <inventory/barcode/conduct-adjustment>` as
```

normal.

Show quantity to count
----------------------

When conducting an inventory count, the expected quantity of products is displayed by default, to
provide the user with a baseline to use when performing the count. However, as this can result in
users relying on this count instead of performing a new count, this quantity can be hidden.

> Navigate to :menuselection:`Inventory app --> Configuration --> Settings`. Clear the :guilabel:`Show
> Quantity to Count` checkbox, then click :guilabel:`Save`.

Reviewer 3: ```suggestion
Navigate to :menuselection:`Inventory app --> Configuration --> Settings`. In the :guilabel:'Barcode' section, clear the :guilabel:`Show
Quantity to Count` checkbox, then click :guilabel:`Save`.
```

.. image:: adjustments/show-quantity-to-count-disabled.png
   :alt: An inventory count without the show quantity to count feature enabled.

.. _inventory/barcode/conduct-adjustment:

Conducting an inventory adjustment
==================================

Navigate to the :menuselection:`Barcode app --> Inventory count`.

.. image:: adjustments/adjustments-barcode-scanner.png
   :alt: Barcode app start screen with scanner.

To begin the adjustment, first scan the *source location*, which is the current location in the
warehouse of the product whose count should be adjusted. Then, scan the product barcodes.

.. tip::
   If the warehouse *multi-location* feature is **not** enabled in the database, a source location
   does not need to be scanned. Instead, scan the product barcode to start the inventory adjustment.

Change the quantity of a product
--------------------------------

> The quantity of a product can be changed in multiple ways during an adjustment.

Reviewer 3: Is it just these two ways? if so, I'd say "two ways" and then you can list them in either bullets or numbers.

add or subtract quantity of the product, and the number keys can be used to add quantity, as well.

.. example::
   In the below inventory adjustment, the source location `WH/Stoc/Shelf 1` was scanned, assigning
   the location. Then, the barcode for the product `[FURN_7888] Desk Stand with Screen` was scanned
   three times, increasing the units in the adjustment. Additional products can be added to this
   adjustment by scanning the barcodes for those specific products.

   .. image:: adjustments/adjustments-barcode-inventory-client-action.png
      :alt: Barcode Inventory Client Action page with inventory adjustment.

Finalize the adjustment
-----------------------

After counting all of the products, review the entries to ensure all the counted quantities are
accurately entered. To complete the inventory adjustment, click :guilabel:`Apply`.

.. tip::
   The :guilabel:`Validate` barcode can be scanned in place of clicking the :guilabel:`Apply`
   button.

Odoo then navigates back to the :guilabel:`Barcode Scanning` screen. A small green banner appears in
the top-right corner, confirming the inventory count has been updated.

> Manually add products to inventory adjustment

Reviewer 3: ```suggestion
Manually add products to inventory adjustment
```
Manually add products to inventory adjustments
or
Manually add products to an inventory adjustment

=============================================

> When the barcodes for the location or product are not available, Odoo **Barcode** can still be used
> to perform inventory adjustments.

Reviewer 3: ```suggestion
When barcodes for location or products are not available, Odoo **Barcode** can still be used
to perform inventory adjustments.
```

To do this, navigate to the :menuselection:`Barcode app --> Barcode Scanning --> Inventory

> Adjustments`.

Reviewer 3: Is this correct? I see 'Barcode' app then the prompt for 'Operations' or 'Inventory Count'. Not sure exactly which - I assume counts.

Reviewer 3: Looks like 'Operations' > 'Internal transfers'?

Once ready, click :guilabel:`Confirm` to confirm the changes.

To apply the inventory adjustment, click :guilabel:`Apply`.

Once applied, Odoo navigates back to the :guilabel:`Barcode Scanning` screen. A small green banner
appears in the top-right corner, confirming validation of the adjustment.

Requested inventory counts
==========================

After an inventory count is :ref:`assigned <inventory/plan-counts>`, they can be accessed through
the **Barcode** application. To view a requested inventory count, navigate to the
:menuselection:`Barcode app` dashboard. If a count has been requested, the number of products to be
counted is listed on the :guilabel:`Inventory Count` button.

.. image:: adjustments/assigned-count.png

>    :alt: The Barcode dashboard with an assigned count.

Reviewer 3: If this is the end, I'd wrap it up with a 'click the 'Inventory Count' button and proceed with the count as usual (or you can link to the directions in the section right above).

Once ready, click :guilabel:`Confirm` to confirm the changes.

To apply the inventory adjustment, click :guilabel:`Apply`.

Once applied, Odoo navigates back to the :guilabel:`Barcode Scanning` screen. A small green banner
appears in the top-right corner, confirming validation of the adjustment.

Requested inventory counts
==========================

After an inventory count is :ref:`assigned <inventory/plan-counts>`, they can be accessed through
the **Barcode** application. To view a requested inventory count, navigate to the
:menuselection:`Barcode app` dashboard. If a count has been requested, the number of products to be
counted is listed on the :guilabel:`Inventory Count` button.

.. image:: adjustments/assigned-count.png

>    :alt: The Barcode dashboard with an assigned count.

Reviewer: Oh wait you do have an assign inventory count section! I think this should be moved up higher?
